### DISCUSSION TITLE
Device groups

### DISCUSSION THREAD

**Discussion by goeran03** [arendst/Tasmota#13514] [Q&A] [answered]
How to use device groups with following situation:
1x shelly i3 with 2 inputs used
2x shelly 1PM with hardware input used

input 1 of i3 and 1st 1PM shall turn off/on relais of 1st 1PM
input 2 of i3 and 2nd 1PM shall turn off/on relais of 2nd 1PM

I tried with 2 device groups. 
1st group=I3 and 1st 1PM
2nd group= I3 and 2nd 1PM
Here the "DevGroupSend" command works fine. but switches did not on 2nd 1PM

I tried then creating virtual (not really wired relais) on 2nd 1PM and on I3 as well in that way that relais Num1 turnes physically on/off lamp1 and relais Num2 on 2nd 1PM (has  virtual relais Num1)  switches physically lamp2. This works quit good except if you turn both switches on I3 simultanously (undefined result).

Any idea here?

G.

**Comment by barbudor** [comment]
May be you can share what configuration you've done on each device.
You shouldn't need virtual relays if you are using proper named dev groups

**Reply by goeran03** [reply]
shelly 1PM num1:
{"SetOption85":"ON"}{"DevGroupName1":"KUE_L","DevGroupName2":"","DevGroupName3":"","DevGroupName4":""}
{"DevGroupShare":{"In":"FFFFFFFF","Out":"FFFFFFFF"}}
{"DevGroupStatus":{"Index":0,"GroupName":"KUE_L","MessageSeq":2,"MemberCount":1,"Members":[{"IPAddress":"192.168.179.83","ResendCount":0,"LastRcvdSeq":3,"LastAckedSeq":2}]}}

shelly 1PM num2:
{"SetOption85":"ON"}{"DevGroupName1":"KUE_R","DevGroupName2":"","DevGroupName3":"","DevGroupName4":""}
{"DevGroupShare":{"In":"FFFFFFFF","Out":"FFFFFFFF"}}
{"DevGroupStatus":{"Index":0,"GroupName":"KUE_R","MessageSeq":2,"MemberCount":1,"Members":[{"IPAddress":"192.168.179.83","ResendCount":0,"LastRcvdSeq":11,"LastAckedSeq":2}]}}

shelly i3
{"SetOption85":"ON"}{"DevGroupName1":"KUE_L","DevGroupName2":"KUE_R","DevGroupName3":"","DevGroupName4":""}
{"DevGroupShare":{"In":"FFFFFFFF","Out":"FFFFFFFF"}}
"DevGroupStatus":{"Index":0,"GroupName":"KUE_L","MessageSeq":3,"MemberCount":1,"Members":[{"IPAddress":"192.168.179.90","ResendCount":0,"LastRcvdSeq":2,"LastAckedSeq":3}]}}
"DevGroupStatus":{"Index":1,"GroupName":"KUE_R","MessageSeq":11,"MemberCount":1,"Members":[{"IPAddress":"192.168.179.89","ResendCount":1,"LastRcvdSeq":2,"LastAckedSeq":11}]}}

hope this is clear so far
:)

**Comment by pcdiem** [comment]
Do you have SO88 1 on shelly i3 so the relay are put in separate device groups?

**Reply by goeran03** [reply]
no setoption88 is off on all devices

**Reply by goeran03** [reply]
with so88 1 on shelly i3  both switches control the same shelly 1PM

when I configure additional RELAIS1 and 2 (as default there are only SWITCH1/2 ) everything is fine :)

**Comment by pcdiem** [comment]
I'm not familiar with how the shelly i3 works. From what I read, there's only one relay and no buttons/switches. Is it only controlled via software? Does the Tasmota web UI have three On/Off buttons on the home page?

**Reply by goeran03** [reply]
no physcal relais at all on I3
but obviously it as only one POWER state

**Comment by pcdiem** [comment]
What do you do when you want it to turn on shelly 1PM num2?

**Reply by goeran03** [reply]
now I have 2 toggle switches on I3 (only when RELAISs are configured) 
butten1 turnes off/on shelly num1 and
button2 does so on shelly2
same with input signals
so actually I'm fine now, but still thinking it should be possible with no unnecssary RELAIS configuration.

**Reply by goeran03** [reply]
The I3 has no buttons and no relays.
It has only three input signals controlling its switch -states.
It is possible using rules to trigger MQTT publishings which can be used to to anything. So I did in the past. Let them talk directly to each other with no not MQTT broker in chain seems to be a  more smart approach.

thanks for your help

**Comment by pcdiem** [comment, marked as answer]
Device groups is sending/receiving the relay state changes so I would think you'd need two relays configured. The only other way I can think of would be to use a rule that acts on switch 2 toggle and uses DevGroupSend2 to send that toggle to device group 2.

**Reply by pcdiem** [reply]
With SO88 0, device groups sends the state of all relays to device group 1. With SO88 1, it sends the state of relay 1 to device group 1, the state of relay 2 to device group 2, etc.

**Reply by goeran03** [reply]
oh now I understand thanks

**Reply by goeran03** [reply]
could be an idea for new feature: dealing also with SWITCH states in groups

**Reply by pcdiem** [reply]
The PWM Dimmer module does this but it's meant for wall switches and includes a lot of other functions such as dimming on holds.

**Reply by barbudor** [reply]
@goeran03 are you using rocker switches or push buttons?
If using push buttons you can define Buttons in the template

**Reply by goeran03** [reply]
no using rocker switches
